src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -359,9 +404,26 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
                     currentList._memoizedOperation.Apply(items, currentGlobsToIgnore);
                 }
 
+                ProcessNonWildCardItemUpdates(itemsWithNoWildcards, items);
+
                 return items;
             }
 
+            private static void ProcessNonWildCardItemUpdates(Dictionary<string, UpdateOperation> itemsWithNoWildcards, ImmutableList<ItemData>.Builder items)
+            {
+                if (itemsWithNoWildcards.Count > 0)
+                {
+                    for (int i = 0; i < items.Count; i++)
+                    {
+                        if (itemsWithNoWildcards.TryGetValue(FileUtilities.GetFullPath(items[i].Item.EvaluatedInclude, items[i].Item.ProjectDirectory), out UpdateOperation op))
+                        {
+                            items[i] = op.UpdateItem(items[i]);
```

Comment:
Does applying the operation here prevent it from eventually getting called on line 404?

Some comments in the code would really make things easier to review, this area in particular seems to be lacking some comments. If you have a fresh understanding of some of these functions/classes (ie, MemoizedOperation, what `currentList._memoizedOperation.Apply(items, currentGlobsToIgnore);` does vs `op.UpdateItem(items[i])` and how they're similar/different they would be **greatly** appreciated.

Reply:
Sorta, but it's a little more complicated. (I'm sure that's exactly what you wanted to hear, right? 😄 )

If it's determined that the operation doesn't have things that expand like wildcards, it will hit the continue statement on line 386. That prevents line 404 from running for that operation. Additionally, at that point, it would have been "addedToBatch," meaning it's in the dictionary that will be passed to this function next time we determine that we can't just add more. This function should always be called if the dictionary is nonempty at such a point (or, in reality, at all other times, but it does ~nothing if it's empty), so the real dependency here is that if an item has no fragments with wildcards, and it's an update operation, it will eventually get here, and line 404 doesn't run, but they don't affect each other directly.

N.B.: One of the restrictions here is that it only affects UpdateOperations. That's mostly because that's the easy way to do it—they get replaced in a 1:1 way. Other operations (Remove) could be included as well, but I wanted to try to keep this fairly simple.